virtual/environment.go

```diff
@@ -847,18 +893,52 @@ func (r *environment) isClosed() bool {
 	return r.shutdownState.closed
 }
 
+// pickServerForInvocation selects a server for invocation based on the provided references and retry policy.
+// It returns the selected actor reference, its index in the references slice, and an error if no references are available.
+// Note: The function may change the order of items in the references slice for efficiency and avoiding unnecessary copies.
 func (r *environment) pickServerForInvocation(
 	references []types.ActorReference,
 	create types.CreateIfNotExist,
-) (types.ActorReference, error) {
-	// TODO: implement invokation strategies in 'create' e.g. region-based, multi-invoke...
+) (types.ActorReference, int, error) {
 	if len(references) == 0 {
-		return types.ActorReference{}, fmt.Errorf("no references available")
+		return types.ActorReference{}, 0, fmt.Errorf("no references available")
+	}
+
+	switch create.Options.ReplicationStrategy {
+	case types.ReplicaSelectionStrategySorted:
+		// Sort the references slice based on the server ID in descending order.
+		// This way, the retry selection is biased towards the replica with the highest server ID over time.
+		result, idx := findExtreme(references, func(i, j int) bool {
+			return references[i].Physical.ServerID > references[j].Physical.ServerID
+		}, true)
+		return result, idx, nil
+	case types.ReplicaSelectionStrategyRandom:
+		fallthrough
+	default:
+		// Use a random selection strategy by generating a random index within the range of available references.
+		// This evenly distributes the retry selection among the available replicas.
+		r.randState.Lock()
+		defer r.randState.Unlock()
+		idx := r.randState.rng.Intn(len(references))
+		return references[idx], idx, nil
+	}
+}
+
+// findExtreme finds the extreme element (min or max) in the slice based on the provided comparator function.
```

Comment:
lol this feels a bit over the top. How about just a plain min or max function over strings with no generics?